### DISCUSSION TITLE
RecursionField里面用的schema采用usememo浅比较，现有动态修改shcema的需求，有什么好方案实现schema动态变更？

### DISCUSSION THREAD

**Discussion by wittech** [alibaba/formily#2823] [Q&A 2.x]
const fieldSchema = useMemo(() => new Schema(props.schema), [props.schema])
这里除了改用useDeepCompareMemo外，还有别的做法吗？

**Comment by laibao101** [comment]
换个思路，直接就让schema整个引用地址都变化呢？这样浅比较也能动态变更了。不过缺陷也很明显。
